### PR TITLE
A few fixes to SFM.

### PR DESCRIPTION
**Pull request by arokem**
Previous reference was to a preprint that is now superseded by a peer-reviewed publication.

### CODE REVIEW

**Comment by arokem** [comment]
Changed this to WIP, because I need to do a couple more things in here.

**Comment by arokem** [comment]
This is ready for review. I managed to remove many of the warnings that arise here. The remaining ones would require some bending over backward, that I would rather avoid, because it would make the code hard to read, or slower to run (unless someone has some good suggestion...).

File: dipy/reconst/sfm.py [hunk 1]
```diff
@@ -352,12 +367,20 @@ def __init__(self, gtab, sphere=None, response=[0.0015, 0.0005, 0.0005],
 
     @auto_attr
     def design_matrix(self):
+        """
+        The design matrix for a SFM.
+
+        Returns
+        -------
+        ndarray : each column is a rotated version of the response function.
```
Comment (Garyfallidis) [inline dipy/reconst/sfm.py]: Non standard docstring for parameter explanation

**Comment by arokem** [comment]
OK - I have addressed your comment.

**Comment by arokem** [comment]
And rebased.

**Comment by arokem** [comment]
Now rebased again, to drop the 3.2 Travis build that was failing.

**Comment by arokem** [comment]
Rebased again. Could someone please take a look?

**Pull request merged**